tools/autograd/gen_autograd.py

```diff
@@ -23,7 +23,15 @@
 
 import argparse
 import os
+# Favoring from...to instead of import...as
+# due to a discrepancy between them up to Python version 3.6
+# https://stackoverflow.com/questions/24807434/imports-in-init-py-and-import-as-statement
+from tools.codegen.api import cpp
+from tools.codegen.api.autograd import *
```

Comment:
nit: I am not a fan of `*` here, can you name explicitly the objects you actually use?